R/help.R

```diff
@@ -238,33 +246,63 @@ help_formals_handler.python.builtin.object <- function(topic, source) {
   NULL
 }
 
+sphinx_doc_params_pattern <- function() { "\nParameters\n+[-]+" }
+
+sphinx_doc_params_matches <- function(doc) {
+  regexpr(pattern = sphinx_doc_params_pattern(), doc)
+}
+
+is_sphinx_doc <- function(doc) {
+  sphinx_doc_params_matches(doc)[[1]] != -1
+}
+
+sphinx_doctree_from_doc <- function(doc) {
+  module_name <- "docutils"
+  if (py_module_available(module_name)) {
+    docutils <- import(module_name)
+    docutils$core$publish_doctree(doc)
+  } else {
+    stop(paste0(module_name, " module needs to be installed to extract doc from Sphinx style documentation."))
```

Comment:
This is probably `NULL` instead of `stop()`. But is there a way handle this in help handlers so no help is shown in this case instead of failing for users?  @jjallaire

Reply:
Shouldn't a return of `NULL` effectively do that? The help handler does that early on when checking for e.g. Python availability.

Is there some sane fallback we can use for Sphinx documentation when `docutils` is not available? (Does the default help handler do something sensible in that case?)

Reply:
I will have to test that out. If the default help handler does something sensible, should we use the default help handler as the fallback when 1) docutils is not available or 2) any exceptions happen during the runtime of sphinx help handler?

Reply:
I would vote for both cases just so the user always gets some help (but hopefully perceptive users will notice if / when Sphinx documentation isn't show as they except and let us know)

Reply:
So we don’t display any message at all? I am still not sure how to display it if needed anyways.

Reply:
Just verified the default help handlers (before this PR) gives the following for functions documented in Sphinx style:

![screen shot 2018-03-01 at 5 25 17 pm](https://user-images.githubusercontent.com/4269898/36873130-8cc07746-1d75-11e8-86e4-6eaf8cc595fe.png)

So I believe it is good to just fallback to the default ones if any of 1) or 2) happens. 

@jjallaire @kevinushey Could you confirm or let me know if there's better way to handle this? Also, what's the best way to let users know that they should install `docutils` module?

Reply:
Two ideas:

a) Display the note "...need to install docutils..." under each parameter description.
b) A warning message for "...need to install docutils..." when the user is trying to look it up.

Reply:
I don't think a user will know what to do with "need to install docutils" so this seems close to a message that will mostly be frustrating rather than helpful. Do we not expect that this will be commonly available?

Reply:
It maybe available through some conda setup or through other dependencies.We could give them better instructions, e.g. telling them to `pip install docutils` or give them a method similar to `install_tensorflow()` or `install_keras()`. 

Or we could just simply fallback and (as Kevin said) hopefully perceptive users will notice if / when Sphinx documentation isn't shown as they except and let us know.